Rakefile

```diff
@@ -1,29 +1,124 @@
+  # docs:yard task
+  YARD::Rake::YardocTask.new
+
+  desc "Clean/remove the generated YARD Documentation cache"
+  task :clean do
+    rakefile_path = File.expand_path(File.dirname(__FILE__))
+    FileUtils.rm_rf(File.join(rakefile_path,YARD_DIR))
+  end
+
+  desc "Tell me about YARD undocummented objects"
+  YARD::Rake::YardocTask.new(:undoc) do |t|
+    t.stats_options = ["--list-undoc"]
+  end
+
+  desc "Measure YARD coverage"
+  require "yardstick/rake/measurement"
+  Yardstick::Rake::Measurement.new(:measure) do |measurement|
+    measurement.output = "yardstick/report.txt"
+  end
+
+  desc "Verify YARD coverage"
+  require "yardstick/rake/verify"
+  config = {"require_exact_threshold" => false}
+  Yardstick::Rake::Verify.new(:verify, config) do |verify|
+    verify.threshold = 57 # FIXME
+  end
+
+  desc "Generate static project architecture graph. (Calls docs:yard)"
+  # this calls `yard graph` so we can"t use the yardoc tasks like above
+  #   We could create a YARD:CLI:Graph object.
+  #   But we still have to send the output to the graphviz processor, etc.
+  task arch: [:yard] do
+    # rake can be run from any dir under here
+    #   so we don't know where we are
+    #   yard graph needs access to the lib file tree, and you can't specify it
+    #   to yard.  so we need to chdir here.
+    original_dir = Dir.pwd
+    Dir.chdir(File.expand_path(File.dirname(__FILE__)))
+    graph_processor = "dot"
+    if exe_exists?(graph_processor)
+      FileUtils.mkdir_p(DOCS_DIR)
+      if system("yard graph --full | #{graph_processor} -Tpng " \
+          "-o #{DOCS_DIR}/arch_graph.png")
+        puts "we made you a class diagram: #{DOCS_DIR}/arch_graph.png"
+      end
+    else
+      puts "ERROR: you don't have dot/graphviz; punting"
+    end
+    Dir.chdir(original_dir)
+  end
+end
 
 namespace :test do
+  desc "check number of lines of code changed. To protect against long PRs"
+  task "diff_length" do
+    max_length = 150
+    target_branch = ENV["DISTELLI_RELBRANCH"] || "master"
+    diff_cmd = "git diff --numstat #{target_branch}"
+    sum_cmd  = "awk '{s+=$1} END {print s}'"
+    cmd      = "[ `#{diff_cmd} | #{sum_cmd}` -lt #{max_length} ]"
+    puts "checking if diff length is less than #{max_length} LoC"
+    exit system(cmd)
+  end
 
-  desc "Run unit tests"
-  RSpec::Core::RakeTask.new('unit') do |t|
-    t.rspec_opts = ['--color']
-    t.pattern = ENV['SPEC_PATTERN']
+  RuboCop::RakeTask.new do |task|
+    task.options = ["--debug"]
   end
 
-  task :unit => [:check_unit]
+  allowed_complexity = 585 # <cough!>
+  FlogTask.new :flog, allowed_complexity, %w[lib]
+  allowed_repitition = 0
+  FlayTask.new :flay, allowed_repitition, %w[lib]
+  RoodiTask.new
+  RubyCritic::RakeTask.new
+  RubyCritic::RakeTask.new do |task|
+    task.paths   = FileList['lib/**/*.rb']
+  end
+
+  begin
+    require "rspec/core/rake_task"
+    RSpec::Core::RakeTask.new(:spec)
+    desc "Run unit tests"
+    RSpec::Core::RakeTask.new do |t|
```

Comment:
What does this second new do?  What task does it create?

Reply:
actually, i think the first one is a duplication bug.  or really does nothing and is overwritten by the second which has some other config args.  i think i stripped this out in the gem_of version